compiler/codegen.js

```diff
@@ -2322,9 +2322,69 @@ const generateVar = (scope, decl) => {
   const global = topLevel || decl._bare;
 
   for (const x of decl.declarations) {
-    const name = mapName(x.id.name);
+    if (x.id.type == "ArrayPattern") {
+      const decls = [];
+      const tmpName = "array_" + randId();
+      let i = 0;
+      for (const e of x.id.elements) {
+        if (e.type == "RestElement") {
+          decls.push({
+            type: "VariableDeclarator",
+            id: { type: "Identifier", name: e.argument.name },
+            init: {
+              type: "CallExpression",
+              callee: {
+                type: 'Identifier',
+                name: '__Array_prototype_slice'
+              },
+              arguments: [
+                { type: "Identifier", name: tmpName },
+                { type: "Literal", value: i },
+                {
+                  type: "MemberExpression",
+                  object: { type: "Identifier", name: tmpName, },
+                  property: { type: "Identifier", name: "length", }
+                }
+              ]
+            }
+          });
+          break;
+        }
+        decls.push({
+          type: "VariableDeclarator",
+          id: { type: "Identifier", name: e.name },
+          init: {
+            type: "MemberExpression",
+            object: { type: "Identifier", name: tmpName, },
+            property: { type: "Literal", value: i }
+          }
+        });
+        i++;
+      }
+
+      return [
+        ...generateVar(scope, {
```

Comment:
Likely best to append to a `declarations` array declared at the top of this function as `[...decl.declarations]` or so instead of calling ourselves again

Reply:
It could yeah actually. Seems unneeded for now so I think we can just stick with this.